samples/snippets/csharp/how-to/strings/SearchStrings.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace HowToStrings
+{
+    public class SearchStrings
+    {
+        public static void Examples()
+        {
+            SearchWithMethods();
+            RegularExpressionsOne();
+            RegularExpressionsValidation();
+        }
+
+        public static void SearchWithMethods()
+        {
+            // <Snippet1>
+            string factMessage = "Extension methods have all the capabilities of regular static methods.";
+
+            // Write the string and include the quotation marks.
+            Console.WriteLine($"\"{factMessage}\"");
+
+            // Simple comparisons are always case sensitive!
+            bool containsSearchResult = factMessage.Contains("extension");
+            Console.WriteLine($"Starts with \"extension\"? {containsSearchResult}");
+
+            // For user input and strings that will be displayed to the end user, 
+            // use the StringComparison parameter on methods that have it to specify how to match strings.
+            bool ignoreCaseSearchResult = factMessage.StartsWith("extension", System.StringComparison.CurrentCultureIgnoreCase);
+            Console.WriteLine($"Starts with \"extension\"? {ignoreCaseSearchResult} (ignoring case)");
+
+            bool endsWithSearchResult = factMessage.EndsWith(".", System.StringComparison.CurrentCultureIgnoreCase);
+            Console.WriteLine($"Ends with '.'? {endsWithSearchResult}");
+
+            /*
+                Output:
+                "Extension methods have all the capabilities of regular static methods."
+                Starts with "extension"? False
+                Starts with "extension"? True (ignoring case)
+                Ends with '.'? True
+            */
+
+            // </Snippet1>
+        }
+
+        private static void SearchByIndex()
```

Comment:
Curious: why is this method the only one that is private and not called from the `Examples()` method?

Reply:
Good catch. They were all meant to be private, and all should be called.